Fix hiding navigation on root folder

Nitwel opened the pull request:
Fixes #19493 

![Screenshot 2023-09-15 at 03-36-43 Directus](https://github.com/directus/directus/assets/5363448/91be6b4b-7a37-438b-86e2-f078f73f5c23)

Nitwel committed:
fix hiding navigation on root folder

paescuj reviewed (dismissed):
We need to introduce new options to address this - simply showing the navigation in every case would be a breaking change, see https://github.com/directus/directus/issues/19493#issuecomment-1691575197.

paescuj commented:
Closing for now in favor of new pull request which addresses https://github.com/directus/directus/pull/19518#issuecomment-1691656913.

Nitwel committed:
Add root folder and showNavigation options

Nitwel commented:
Added the option now to show or hide the drawer manually. And when the root folder is set, it properly only shows that folder and it's children.

Also added tests for the useFiles composable. 🧪 

![grafik](https://github.com/directus/directus/assets/22577866/98f2f2a4-95f4-4d41-9751-335b3272fade)

![grafik](https://github.com/directus/directus/assets/22577866/ae2958f1-8aca-47e7-a786-dff8ed9e4460)

paescuj committed:
Merge branch 'main' into fix-19493

paescuj committed:
Format files

DanielBiegler reviewed (dismissed):
There seems to be a problem with refetching the folders after creating/deleting/moving folders

[Screencast from 04.09.2023 15:37:27.webm](https://github.com/directus/directus/assets/14810858/b793fa24-a5a5-4650-9171-aa668542d71a)

Switched back to main branch for comparison and dont have the problem there.

DanielBiegler committed:
remove unused imports

DanielBiegler committed:
make linter happy

paescuj committed:
Merge branch 'main' into fix-19493

paescuj committed:
Merge branch 'main' into fix-19493

paescuj committed:
Fix useFolders issue, always show folder nav, fix tedious type issues

paescuj committed:
Add changeset

paescuj committed:
Merge branch 'main' into fix-19493

paescuj committed:
Update app/src/views/private/components/files-navigation.vue

Co-authored-by: Azri Kahar <42867097+azrikahar@users.noreply.github.com>

paescuj commented on app/src/views/private/components/files-navigation.vue:
```diff
@@ -84,12 +86,18 @@ const props = defineProps<{
+const rootFolderInfo = computed(() => {
+	if (!folders.value || !props.rootFolder) return;
+
+	return folders.value.find((folder) => folder.id === rootFolder!.value);
```
Great catch ❤️
